server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -252,24 +252,30 @@ public void write(OutputStream outputStream) throws WebApplicationException
             )
             .header("X-Druid-Query-Id", queryId);
 
-        if (responseContext.get(ResponseContext.CTX_ETAG) != null) {
-          builder.header(HEADER_ETAG, responseContext.get(ResponseContext.CTX_ETAG));
-          responseContext.remove(ResponseContext.CTX_ETAG);
+        if (responseContext.get(ResponseContext.Key.ETAG) != null) {
+          builder.header(HEADER_ETAG, responseContext.get(ResponseContext.Key.ETAG));
```

Comment:
I think would be clearer to call this variable `responseBuilder`

Reply:
Renamed